x2go fails to build on master

## Issue description

```
[build CXX] svx/source/engine3d/helperhittest3d.cxx
g++ -c -O2 -Wall -W -D_REENTRANT -fPIC -DCFGCLIENT -D__linux__ -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_X11EXTRAS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../x2goclient-4.1.1.1 -I. -I../src -isystem /nix/store/wh2jz3by2gd647mxz0x0havksrqqabhh-qtsvg-5.11.0-dev/include/QtSvg -isystem /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets -isystem /nix/store/4w9dq4g9h09yc69k04xgw7ggf6kg1vvf-qtx11extras-5.11.0-dev/include/QtX11Extras -isystem /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtGui -isystem /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtNetwork -isystem /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtCore -I. -I/nix/store/m5b6n9762kyrmmvvdq54ba8aw05sfxly-mesa-noglu-18.0.3-dev/include -I. -I/nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/mkspecs/linux-g++ -o printwidget.o ../src/printwidget.cpp
[ 79%] Building CXX object plugins/impex/csv/CMakeFiles/kritacsvexport.dir/csv_saver.cpp.o
[build CXX] svx/source/engine3d/helperminimaldepth3d.cxx
../src/printwidget.cpp: In constructor 'PrintWidget::PrintWidget(QWidget*)':
../src/printwidget.cpp:77:36: error: invalid use of incomplete type 'class QButtonGroup'
  QButtonGroup* bg=new QButtonGroup();
                                    ^
In file included from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qcheckbox.h:44:0,
                 from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/QCheckBox:1,
                 from ./ui_printwidget.h:14,
                 from ../src/printwidget.h:22,
                 from ../src/printwidget.cpp:18:
/nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'class QButtonGroup'
 class QButtonGroup;
       ^~~~~~~~~~~~
../src/printwidget.cpp:78:4: error: invalid use of incomplete type 'class QButtonGroup'
  bg->addButton ( ui.rbPrint );
    ^~
In file included from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qcheckbox.h:44:0,
                 from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/QCheckBox:1,
                 from ./ui_printwidget.h:14,
                 from ../src/printwidget.h:22,
                 from ../src/printwidget.cpp:18:
/nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'class QButtonGroup'
 class QButtonGroup;
       ^~~~~~~~~~~~
../src/printwidget.cpp:78:6: error: invalid use of incomplete type 'class QButtonGroup'
  bg->addButton ( ui.rbPrint );
      ^~~~~~~~~
In file included from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qcheckbox.h:44:0,
                 from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/QCheckBox:1,
                 from ./ui_printwidget.h:14,
                 from ../src/printwidget.h:22,
                 from ../src/printwidget.cpp:18:
/nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'class QButtonGroup'
 class QButtonGroup;
       ^~~~~~~~~~~~
../src/printwidget.cpp:79:4: error: invalid use of incomplete type 'class QButtonGroup'
  bg->addButton ( ui.rbView );
    ^~
In file included from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qcheckbox.h:44:0,
                 from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/QCheckBox:1,
                 from ./ui_printwidget.h:14,
                 from ../src/printwidget.h:22,
                 from ../src/printwidget.cpp:18:
/nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'class QButtonGroup'
 class QButtonGroup;
       ^~~~~~~~~~~~
../src/printwidget.cpp:79:6: error: invalid use of incomplete type 'class QButtonGroup'
  bg->addButton ( ui.rbView );
      ^~~~~~~~~
In file included from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qcheckbox.h:44:0,
                 from /nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/QCheckBox:1,
                 from ./ui_printwidget.h:14,
                 from ../src/printwidget.h:22,
                 from ../src/printwidget.cpp:18:
/nix/store/ni5sy11fvw9yj5i8bihbvxprm2cy6d3q-qtbase-5.11.0-dev/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'class QButtonGroup'
 class QButtonGroup;
       ^~~~~~~~~~~~
make[1]: *** [Makefile:1465: printwidget.o] Error 1
make[1]: Leaving directory '/tmp/nix-build-x2goclient-4.1.1.1.drv-0/x2goclient-4.1.1.1/client_build'
make: *** [Makefile:76: build_client] Error 2
builder for '/nix/store/fjf0j2r7biyrhm71a4b3jdcz2yby6v7y-x2goclient-4.1.1.1.drv' failed with exit code 2
cannot build derivation '/nix/store/zfm3sfns8gys72yxy953vn28icwqlxga-system-path.drv': 1 dependencies couldn't be built
building '/nix/store/ii3sihc8cnz4ih3n9c0p47wv0qq7al1s-xkb-validated.drv'...
building '/nix/store/61b3ymf6r0i5i4kd8kim5hzzjhj8wvwh-xlock.pam.drv'...
building '/nix/store/hp1pjm9rs9v4qas648hyxsx7qcqkyyjq-xscreensaver.pam.drv'...
building '/nix/store/4djr7pc2p4l20w4481966c8v8yz20y93-xserver.conf.drv'...
cannot build derivation '/nix/store/pmfwar666j264aysa2ha0nqwh5gddvkh-nixos-system-nixos-18.09.git.6916565.drv': 1 dependencies couldn't be built
error: build of '/nix/store/pmfwar666j264aysa2ha0nqwh5gddvkh-nixos-system-nixos-18.09.git.6916565.drv' failed
```

### Steps to reproduce



## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

--- a/pkgs/applications/networking/remote/x2goclient/default.nix
+++ b/pkgs/applications/networking/remote/x2goclient/default.nix
@@ -3,11 +3,11 @@ makeWrapper, qtbase, qtsvg, qtx11extras, qttools, phonon }:
 
 stdenv.mkDerivation rec {
   name = "x2goclient-${version}";
-  version = "4.1.1.1";
+  version = "4.1.2.0";
 
   src = fetchurl {
     url = "http://code.x2go.org/releases/source/x2goclient/${name}.tar.gz";
-    sha256 = "0jzlwn0v8b123h5l7hrhs35x2z6mb98zg1s0shqb4yfp2g641yp3";
+    sha256 = "1x1iiyszz6mbrnsqacxzclyx172djq865bw3y83ya7lc9j8a71zn";
   };
 
   buildInputs = [ cups libssh libXpm nxproxy openldap openssh

--- a/pkgs/applications/networking/remote/x2goclient/default.nix
+++ b/pkgs/applications/networking/remote/x2goclient/default.nix
@@ -14,7 +14,9 @@ stdenv.mkDerivation rec {
                   qtbase qtsvg qtx11extras qttools phonon ];
   nativeBuildInputs = [ makeWrapper ];
 
-  patchPhase = ''
+  patches = [ ./qt511.patch ];
+
+  postPatch = ''
      substituteInPlace Makefile \
        --replace "SHELL=/bin/bash" "SHELL=$SHELL" \
        --replace "lrelease-qt4" "${qttools.dev}/bin/lrelease" \

--- a/pkgs/applications/networking/remote/x2goclient/qt511.patch
+++ b/pkgs/applications/networking/remote/x2goclient/qt511.patch
@@ -1,0 +1,15 @@
+diff --git a/src/printwidget.cpp b/src/printwidget.cpp
+index 58a8af7..131d340 100644
+--- a/src/printwidget.cpp
++++ b/src/printwidget.cpp
+@@ -23,6 +23,7 @@
+ #include "x2gosettings.h"
+ #include "x2gologdebug.h"
+ #include <QDir>
++#include <QButtonGroup>
+ #ifdef Q_OS_WIN
+ #include "wapi.h"
+ #endif
+-- 
+2.17.1
+

